packages/angular/projects/ui-angular/src/lib/primitives-poc/view/view-component/view.component.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+  OnChanges,
+  OnInit,
+  Renderer2,
+  SimpleChanges,
+  ViewEncapsulation,
+} from '@angular/core';
+
+/** Config file to get all the configuration related the strings and values */
+import { ComponentSettings } from './ComponentSetting';
+@Component({
+  selector: ComponentSettings.COMPONENTNAME,
+  template: '<ng-content></ng-content>',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AmplifyViewComponent implements OnInit, OnChanges {
+  /** Input Prop for style changes  */
+  @Input() type: 'submit' | 'button' = 'button';
+  @Input() fullWidth: boolean | string = false;
+  @Input() isDisabled: boolean | string = false;
+  @Input() size: 'small' | 'medium' | 'large' = 'medium';
+  @Input() variation: 'primary' | 'default' | 'link' = 'default';
+
+  @HostBinding('type') typeAttr: string;
+  @HostBinding('attr.data-fullwidth') fullWidthAttr: boolean | string;
+  @HostBinding('attr.data-size') sizeAttr: string;
+  @HostBinding('attr.data-variation') variationAttr: string;
+  @HostBinding('style.font-weight') fontWeightAttr: string;
+
+  @HostBinding('attr.disabled') get getDisabled() {
+    return this.isDisabled ? '' : null;
+  }
+
+  constructor(private renderer: Renderer2, private element: ElementRef) {}
+
+  /** onchange method to detecte prop changes and update the accordingly */
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
This is a cool idea! Design question -- what is the engineering benefits of using Renderer2 over `@HostBinding`?

Reply:
For simple elements like `view` though, I still think `Renderer2` is an overkill and can be more simply implemented by just `@HostBinding`.

Renderer2 will be more useful when we implement more complex primitives that doesn't use directive approach. But for `View`, we are using directive approach and so customers will be directly setting HTML elements anyway. I don't see much need to use `Renderer2` to intercept this behavior and set the attributes ourselves (which also defeats the purpose of using directive approach).